include/nlohmann/detail/input/binary_reader.hpp

```diff
@@ -120,6 +124,139 @@ class binary_reader
     }
 
   private:
+
+    /*!
+    @return whether array creation completed
+    */
+    bool get_bson_cstr(string_t& result)
+    {
+        auto out = std::back_inserter(result);
+        while (true)
+        {
+            get();
+            if (JSON_UNLIKELY(not unexpect_eof()))
+            {
+                return false;
+            }
+            if (current == 0x00)
+            {
+                return true;
+            }
+            *out++ = static_cast<char>(current);
+        }
+
+        return true;
+    }
+
+    bool parse_bson_entries(bool is_array)
+    {
+        while (auto entry_type = get())
+        {
+            string_t key;
+            if (!get_bson_cstr(key))
+            {
+                return false;
+            }
+
+            if (!is_array)
+            {
+                sax->key(key);
+            }
+
+            switch (entry_type)
+            {
+                case 0x01: // double
+                {
+                    double number;
+                    get_number<double, true>(number);
+                    sax->number_float(static_cast<number_float_t>(number), "");
+                }
+                break;
```

Comment:
Please put the `break;` into the scope above. This affects all `switch` statements.

Reply:
done